### ISSUE TITLE
MXNet v0.9.3a bug

### ISSUE THREAD

**Issue by kevinthesun** [dmlc/mxnet#5127]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System: Docker Ubuntu 14.04

Package used (Python/R/Scala/Julia): Python

MXNet version: 0.9.3a

Python version and distribution: 2.7.6

## Error Message:
This bug is caught by Yao's notebook automated testing script with Karishma's linear-regression.ipynb notebook. MXNet score mean-square-error metric sometimes returns 'nan'.

## Minimum reproducible example
In mxnet-notebooks/python/tutorials/linear-regression.ipynb cell No.9 and No.10:
```python
#Evaluation
metric = mx.metric.MSE()
model.score(eval_iter, metric)
```
```python
#Evaluation Data
eval_data = np.array([[7,2],[6,10],[12,2]])
eval_label = np.array([11.1,26.1,16.1]) #Adding 0.1 to each of the values 
eval_iter = mx.io.NDArrayIter(eval_data, eval_label, batch_size, shuffle=False)
model.score(eval_iter, metric)
```

Sometimes outputs are 'nan'.

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

Run this notebook several times and print the result. Sometimes it produces desired result, sometimes 'nan'.

## What have you tried to solve it?

If we explictly set 'reset=False", it returns 'nan' every time:
```python
model.score(eval_iter, metric, reset=False)
```
It might be related to NDArrayIte reset.

**Comment by kevinthesun** [comment]
It turns out this issue results from training data and learning rate.
